pcre.go

```diff
@@ -333,12 +337,16 @@ var nullbyte = []byte{0}
 // Match tries to match the specified byte slice to
 // the current pattern by calling Exec and collects the result.
 // Returns true if the match succeeds.
+// Match is a no-op if err is not nil.
 func (m *Matcher) Match(subject []byte, flags int) bool {
+	if m.err != nil {
+		return false
+	}
 	if m.re.ptr == nil {
 		panic("Matcher.Match: uninitialized")
 	}
 	rc := m.Exec(subject, flags)
-	m.matches = matched(rc)
+	m.matches, m.err = matched(rc)
```

Comment:
Whats the reasoning behind having `Matcher` store the error? Seems more intuitive/idiomatic to just have `Match` and `MatchString` return the error directly?

Reply:
wanted to avoid copy-pasting `if err != nil { return err}` everywhere, so following [this post](https://blog.golang.org/errors-are-values), mirrored design of [scanner](https://golang.org/pkg/bufio/#Scanner)